genesyscloud/external_contacts_organization/genesyscloud_externalcontacts_organization_proxy.go

```diff
@@ -0,0 +1,201 @@
+}
+
+// createExternalContactsOrganizationFn is an implementation function for creating a Genesys Cloud external contacts organization
+func createExternalContactsOrganizationFn(ctx context.Context, p *externalContactsOrganizationProxy, externalContactsOrganization *platformclientv2.Externalorganization) (*platformclientv2.Externalorganization, *platformclientv2.APIResponse, error) {
+	externalOrganization, response, err := p.externalContactsApi.PostExternalcontactsOrganizations(*externalContactsOrganization)
+	if err != nil {
+		return nil, response, fmt.Errorf("failed to create external contacts organization: %s", err)
+	}
+
+	return externalOrganization, response, nil
+}
+
+// getAllExternalContactsOrganizationFn is the implementation for retrieving all external contacts organization in Genesys Cloud
+func getAllExternalContactsOrganizationFn(ctx context.Context, p *externalContactsOrganizationProxy, query string) (*[]platformclientv2.Externalorganization, *platformclientv2.APIResponse, error) {
+	var allExternalOrganizations []platformclientv2.Externalorganization
+	const pageSize = 100
+
+	externalOrganizations, response, err := p.externalContactsApi.GetExternalcontactsOrganizations(pageSize, 1, query, nil, "", nil, true)
+	if err != nil {
+		return nil, response, fmt.Errorf("failed to get external organization: %v", err)
+	}
+	if externalOrganizations.Entities == nil || len(*externalOrganizations.Entities) == 0 {
+		return &allExternalOrganizations, response, nil
+	}
+	allExternalOrganizations = append(allExternalOrganizations, *externalOrganizations.Entities...)
+
+	for pageNum := 2; pageNum <= *externalOrganizations.PageCount; pageNum++ {
+		externalOrganizations, _, err := p.externalContactsApi.GetExternalcontactsOrganizations(pageSize, pageNum, query, nil, "", nil, true)
+		if err != nil {
+			return nil, response, fmt.Errorf("failed to get external organization: %v", err)
+		}
+
+		if externalOrganizations.Entities == nil || len(*externalOrganizations.Entities) == 0 {
+			break
+		}
+
+		allExternalOrganizations = append(allExternalOrganizations, *externalOrganizations.Entities...)
+
+	}
+	// Cache the External Contacts resource into the p.externalContactsCache for later use
+	for _, externalOrganization := range allExternalOrganizations {
+		if externalOrganization.Id == nil {
+			continue
+		}
+		rc.SetCache(p.externalOrganizationCache, *externalOrganization.Id, externalOrganization)
+	}
+
+	return &allExternalOrganizations, response, nil
+}
+
+// getExternalContactsOrganizationIdByNameFn is an implementation of the function to get a Genesys Cloud external contacts organization by name
+func getExternalContactsOrganizationIdByNameFn(ctx context.Context, p *externalContactsOrganizationProxy, name string) (id string, retryable bool, apiResponse *platformclientv2.APIResponse, err error) {
+
+	externalOrganizations, response, err := getAllExternalContactsOrganizationFn(ctx, p, name)
```

Comment:
related to comment above. can we have cache lookup for datasource. before making actual api call